util/stmtsummary/v2/stmtsummary.go

```diff
@@ -0,0 +1,504 @@
+	}()
+
+	return ss
+}
+
+// Enabled returns whether the StmtSummary is enabled.
+func (s *StmtSummary) Enabled() bool {
+	return s.optEnabled.Load()
+}
+
+// SetEnabled is used to enable or disable StmtSummary. If disabled, in-memory
+// data will be cleared, (persisted data will still be remained).
+func (s *StmtSummary) SetEnabled(v bool) error {
+	s.optEnabled.Store(v)
+	if !v {
+		s.Clear()
+	}
+
+	return nil
+}
+
+// EnableInternalQuery returns whether the StmtSummary counts internal queries.
+func (s *StmtSummary) EnableInternalQuery() bool {
+	return s.optEnableInternalQuery.Load()
+}
+
+// SetEnableInternalQuery is used to enable or disable StmtSummary's internal
+// query statistics. If disabled, in-memory internal queries will be cleared,
+// (persisted internal queries will still be remained).
+func (s *StmtSummary) SetEnableInternalQuery(v bool) error {
+	s.optEnableInternalQuery.Store(v)
+	if !v {
+		s.ClearInternal()
+	}
+
+	return nil
+}
+
+// MaxStmtCount returns the maximum number of statements.
+func (s *StmtSummary) MaxStmtCount() uint32 {
+	return s.optMaxStmtCount.Load()
+}
+
+// SetMaxStmtCount is used to set the maximum number of statements.
+// If the current number exceeds the maximum number, the excess will be evicted.
+func (s *StmtSummary) SetMaxStmtCount(v uint32) error {
+	if v < 1 {
+		v = 1
+	}
+	s.optMaxStmtCount.Store(v)
+	s.windowLock.Lock()
+	_ = s.window.lru.SetCapacity(uint(v))
+	s.windowLock.Unlock()
+
+	return nil
+}
+
+// MaxSQLLength returns the maximum size of a single SQL statement.
+func (s *StmtSummary) MaxSQLLength() uint32 {
+	return s.optMaxSQLLength.Load()
+}
+
+// SetMaxSQLLength sets the maximum size of a single SQL statement.
+func (s *StmtSummary) SetMaxSQLLength(v uint32) error {
+	s.optMaxSQLLength.Store(v)
+
+	return nil
+}
+
+// RefreshInterval returns the period (in seconds) at which the statistics
+// window is refreshed (persisted).
+func (s *StmtSummary) RefreshInterval() uint32 {
+	return s.optRefreshInterval.Load()
+}
+
+// SetRefreshInterval sets the period (in seconds) for the statistics window
+// to be refreshed (persisted). This may trigger a refresh (persistence) of
+// the current statistics window early.
+func (s *StmtSummary) SetRefreshInterval(v uint32) error {
+	if v < 1 {
+		v = 1
+	}
+	s.optRefreshInterval.Store(v)
+
+	return nil
+}
+
+// Add adds a single stmtsummary.StmtExecInfo to the current statistics window
+// of StmtSummary. Before adding, it will check whether the current window has
+// expired, and if it has expired, the window will be persisted asynchronously
+// and a new window will be created to replace the current one.
+func (s *StmtSummary) Add(info *stmtsummary.StmtExecInfo) {
+	if s.closed.Load() {
+		return
+	}
+	if !s.Enabled() {
+		// StmtSummary is not enabled.
+		return
+	}
+	if info.IsInternal && !s.EnableInternalQuery() {
+		// StmtSummary does not enable internal query statistics
+		// and this record is an internal query.
+		return
+	}
```

Comment:
Can remove this? Since if StmtSummary is disabled, won't run into here. Since it will return early in `func (a *ExecStmt) SummaryStmt(succ bool)`

Reply:
Get it